src/preact-render-spy.js

```diff
@@ -176,14 +238,25 @@ class FindWrapper {
   }
 
   /**
+   * Return an object copy of the attributes from the first node that matched.
+   */
+  attrs() {
+    if (!this[0]) {
+      return null;
+    }
+
+    return Object.assign({}, this[0].attributes);
+  }
+
+  /**
    * Return the text of all nested children concatenated together.
    */
   text() {
     return Array.from(vdomWalk(this.spy.vdomMap, Array.from(this)))
     // Filter for strings (text nodes)
     .filter(value => typeof value === 'string')
     // Concatenate all strings together
-    .reduce((carry, value) => carry + value);
+    .reduce((carry, value) => carry + value, '');
```

Comment:
This fixes a bug. `[].reduce((c, v) => c + v)` produces an error. If you reduce an empty array reduce doesn't have a deterministic value to return. For possibly empty arrays you must start with a value.